snapshot/manifest.go

```diff
@@ -91,6 +92,12 @@ type HasDirEntry interface {
 	DirEntry() *DirEntry
 }
 
+// HasDirEntryFromPlaceholder is implemented by objects that may have a DirEntry
+// stored in the object's corresponding shallow placeholder file.
+type HasDirEntryFromPlaceholder interface {
+	DirEntryFromPlaceholder(ctx context.Context) (*DirEntry, error)
```

Comment:
How about `DirEntry()` or perhaps `DirEntryOrNil()` ?

Reply:
So probably not `DirEntry()` pending the resolution of the comment above? 🙂

I changed `DirEntryFromPlaceholder` to `DirEntryOrNil`.

Aside: maybe `DirEntryFromPlaceholder()` would be better as part of the `fs` package? (Because it needs to be implemented by `localfs` and `ignorefs`) Except that `DirEntry` is in `snapshot` and `snapshot` imports `fs` so `fs` can't import `snapshot`. All the ways to fix this seemed like large source changes.